explorer/explorer.go

```diff
@@ -112,19 +114,24 @@ type Explorer struct {
 
 func (e *Explorer) syncStore(index types.ChainIndex, batchSize int) error {
 	for index != e.cm.Tip() {
-		crus, caus, err := e.cm.UpdatesSince(index, batchSize)
-		if err != nil {
-			return fmt.Errorf("failed to subscribe to chain manager: %w", err)
-		}
+		select {
```

Comment:
```go
for index != e.cm.Tip() && e.ctx.Err() == nil {
    ...
}
return e.ctx.Err()
```
is a little bit cleaner

Reply:
I honestly don't like that pattern. I'd rather have the logic for interrupting the loop isolated in a `select` and then return `nil` to have clear success and interrupt paths rather than mixing them.